### PR TITLE
fix(storage): Remove work manager restricted API usage and upgrade lib.

### PR DESCRIPTION
**Pull request by tylerjroach**
- [ ] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*

*How did you test these changes?*
(Please add a line here how the changes were tested)

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

*General Checklist*
- [ ] Added Unit Tests
- [ ] Added Integration Tests
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by tylerjroach** [comment]
Rather than rely on the internal isStopped (which we can't safely set), we can check if the current coroutine context is active. 

Our workers are CoroutineWorker's. When onStopped is called on a CoroutineWorker the job is marked as inactive (although the work continues to proceed because it wasn't forcefully cancelled). We can, however, check the `isActive` boolean which should be the same thing as onStopped getting called.

File: aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/transfer/worker/BaseTransferWorker.kt [hunk 1]
```diff
@@ -108,11 +107,12 @@ internal abstract class BaseTransferWorker(
 
         return when {
             result.isSuccess -> {
+                Log.d("App", "Success: Coroutine Active ${currentCoroutineContext().isActive}")
```
Comment (mattcreaser) [inline aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/transfer/worker/BaseTransferWorker.kt:110]: Remove this log

File: aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/transfer/worker/BaseTransferWorker.kt [hunk 2]
```diff
@@ -150,8 +150,8 @@ internal abstract class BaseTransferWorker(
         )
     }
 
-    private fun isRetryableError(e: Throwable?): Boolean {
-        return isStopped ||
+    private suspend fun isRetryableError(e: Throwable?): Boolean {
```
Comment (mattcreaser) [inline aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/transfer/worker/BaseTransferWorker.kt:153]: You can move the `isActive` check inline in `doWork` - this function should not need to suspend.

**Review by mattcreaser — APPROVED** [review]
Couple small comments but this looks good!

**Review by mattcreaser — APPROVED** [review]

**Review by gpanshu — APPROVED** [review]

**Pull request merged**